pkg/util/kubectl/kubectl.go

```diff
@@ -354,3 +358,75 @@ func UpdateDeployment(deploy *v1.Deployment) error {
 	_, err := client.AppsV1().Deployments(deploy.Namespace).Update(context.TODO(), deploy, metav1.UpdateOptions{})
 	return err
 }
+
+// PatchOwnerReferenceWithAppInfoFile patch tfjob / pytorchjob ownerReference
+func PatchOwnerReferenceWithAppInfoFile(name, trainingType, appInfoFile, namespace string) error {
+	binary, err := exec.LookPath(kubectlCmd[0])
+	if err != nil {
+		return err
+	}
+	errs := []string{}
+
+	// get training job
+	args := []string{binary, "get", trainingType, name, "--namespace", namespace, "-o json"}
+	cmd := exec.Command("bash", "-c", strings.Join(args, " "))
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		errs = append(errs, string(out))
+	}
+
+	obj := &unstructured.Unstructured{}
+	err = json.Unmarshal(out, obj)
+	if err != nil {
+		return err
+	}
+
+	patch := fmt.Sprintf(`-p='[{"op": "add", "path": "/metadata/ownerReferences", `+
+		`"value": [{"apiVersion": "%s","kind": "%s","name": "%s","uid": "%s","blockOwnerDeletion": true,"controller": true}]}]'`,
+		obj.GetAPIVersion(), obj.GetKind(), name, obj.GetUID())
+
+	data, err := ioutil.ReadFile(appInfoFile)
+	if err != nil {
+		return err
+	}
+	resources := strings.Split(string(data), "\n")
+	// add configmap
+	configmapName := fmt.Sprintf("%v-%v", name, trainingType)
+	resources = append(resources, "configmap/"+configmapName)
+
+	var wg sync.WaitGroup
+	locker := new(sync.RWMutex)
+	for _, r := range resources {
+		resource := r
+		// skip training job.
+		// skip tensorboard service: job operator will add ownerReferences.
+		if strings.Contains(resource, "tfjob.kubeflow.org/") ||
+			strings.Contains(resource, "pytorchjob.kubeflow.org/") ||
```

Comment:
When handling resource objects, the current code uses the strings.Contains function to determine whether to perform the JSON patch operation based on the resource name. This approach may not be accurate. Consider using a more accurate way to determine the resource type, such as using the official Kubernetes API or resource label information.